airflow/providers/cncf/kubernetes/hooks/kubernetes.py

```diff
@@ -363,11 +363,9 @@ def get_custom_object(
     def get_namespace(self) -> Optional[str]:
         """Returns the namespace that defined in the connection"""
         if self.conn_id:
-            connection = self.get_connection(self.conn_id)
-            extras = connection.extra_dejson
-            namespace = extras.get("extra__kubernetes__namespace", "default")
-            return namespace
-        return None
+            return self._get_field('namespace', 'default')
```

Comment:
this is just an optimization since it uses cached prop `conn_extras` whereas `get_namespace` retrieves the conn in every call